### PR TITLE
feat(toolkit): add process documentation

### PR DESCRIPTION
**Pull request by alachambre**
- Add a page that contains the detailed documentation for process definitions and process instances
- Create empty pages for all the references

### CODE REVIEW

File: modules/process-testing/pages/process.adoc [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+[source, Java]
+----
+@BonitaTests(clearBDM = true, deleteProcessInstances = true)
+class ProcessIT {
+
+    private static final String PROCESS_NAME = "myProcess";
+
+    @Test
+    void should_retrieve_process_initiators(BonitaTestToolkit toolkit) {
+        ProcessDefinition processDef = toolkit.getProcessDefinition(PROCESS_NAME);
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]:
```suggestion
        var myProcess= toolkit.getProcessDefinition( "myProcess"); 
```

File: modules/process-testing/pages/process.adoc [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+----
+@BonitaTests(clearBDM = true, deleteProcessInstances = true)
+class ProcessIT {
+
+    private static final String PROCESS_NAME = "myProcess";
+
+    @Test
+    void should_retrieve_process_initiators(BonitaTestToolkit toolkit) {
+        ProcessDefinition processDef = toolkit.getProcessDefinition(PROCESS_NAME); 
+        Set<User> initiators = processDef.getInitiators();
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc:49]:
```suggestion
        Set<User> initiators =  myProcess.getInitiators();
```

File: modules/process-testing/pages/process.adoc [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+----
+@BonitaTests(clearBDM = true, deleteProcessInstances = true)
+class ProcessIT {
+
+    private static final String PROCESS_NAME = "myProcess";
+    private static final String PROCESS_VERSION = "1.0.0";
+
+    @Test
+    void should_retrieve_process_definitions(BonitaTestToolkit toolkit) {
+        ProcessDefinition processDef1 = toolkit.getProcessDefinition(PROCESS_NAME); <1>
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc:29]:
```suggestion
        var myProcess= toolkit.getProcessDefinition("myProcess"); <1>
```

File: modules/process-testing/pages/process.adoc [hunk 4]
```diff
@@ -0,0 +1,253 @@
...
+        ProcessDefinition processDef2 = toolkit.getProcessDefinition(PROCESS_NAME,  PROCESS_VERSION); <2>
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]:
```suggestion
        var myProcessWithExplicitVersion = toolkit.getProcessDefinition("myProcess",  "1.0.0"); <2>
```

File: modules/process-testing/pages/process.adoc [hunk 5]
```diff
@@ -0,0 +1,253 @@
...
+=== Retrieve a process definition
+
+A process definition can be retrieved using the Bonita test toolkit, the process name and optionally the process version. If the process version is not specified then the latest deployed version is retrieved (and not the latest version).
+
+[source, Java]
+----
+@BonitaTests(clearBDM = true, deleteProcessInstances = true)
+class ProcessIT {
+
+    private static final String PROCESS_NAME = "myProcess";
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc:25]: Remove me

File: modules/process-testing/pages/process.adoc [hunk 6]
```diff
@@ -0,0 +1,253 @@
...
+    private static final String PROCESS_VERSION = "1.0.0";
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]: Remove me

File: modules/process-testing/pages/process.adoc [hunk 7]
```diff
@@ -0,0 +1,253 @@
...
+}
+----
+
+=== Start a process
+
+A case can be started from the process definition. It requires a xref:user.adoc[user] to start the process and optionally a xref:contract.adoc[contract]. 
+
+[source, Java]
+----
+@BonitaTests(clearBDM = true, deleteProcessInstances = true)
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]:
```suggestion
@BonitaTests
```
Not relevant for the example above

File: modules/process-testing/pages/process.adoc [hunk 8]
```diff
@@ -0,0 +1,253 @@
...
+class ProcessIT {
+
+    private static final String USER_NAME = "walter.bates";
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]: Avoid using constants for simple example.

File: modules/process-testing/pages/process.adoc [hunk 9]
```diff
@@ -0,0 +1,253 @@
...
+    private static final String PROCESS_NAME_1 = "myProcess1";
+    private static final String PROCESS_NAME_2 = "myProcess2";
+
+    @Test
+    void should_start_process_instances(BonitaTestToolkit toolkit) {
+        User walter = toolkit.getUser(USER_NAME);
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]:
```suggestion
        User walter = toolkit.getUser("walter.bates");
```

File: modules/process-testing/pages/process.adoc [hunk 10]
```diff
@@ -0,0 +1,253 @@
...
+
+        ProcessDefinition processDef1 = toolkit.getProcessDefinition(PROCESS_NAME_1);
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]:
```suggestion
        var myProcess1 = toolkit.getProcessDefinition("myProcess1");
```

File: modules/process-testing/pages/process.adoc [hunk 11]
```diff
@@ -0,0 +1,253 @@
...
+        ProcessInstance processInstance1 = processDef1.startProcessFor(walter); <1>
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc:68]:
```suggestion
        var processInstance1 = myProcess1.startProcessFor(walter); <1>
```

File: modules/process-testing/pages/process.adoc [hunk 12]
```diff
@@ -0,0 +1,253 @@
...
+
+        ProcessDefinition processDef2 = toolkit.getProcessDefinition(PROCESS_NAME_2);
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc]:
```suggestion
        var myProcess2= toolkit.getProcessDefinition("myProcess2");
```

File: modules/process-testing/pages/process.adoc [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+        Contract contract = createContract(); <2>
+        ProcessInstance processInstance2 = processDef2.startProcessFor(walter, contract); <3>
```
Comment (rbioteau) [inline modules/process-testing/pages/process.adoc:72]:
```suggestion
        var processInstance2 = myProcess2.startProcessFor(walter, contract); <3>
```

File: modules/process-testing/pages/process.adoc (hunk 1 above)
Comment (alachambre) [reply modules/process-testing/pages/process.adoc]: I prefer to not use var here, so users can know the type of the object returned by the method.

**Pull request merged**